src/SSHDebugPS/AD7PortSupplier.cs

```diff
@@ -114,39 +115,18 @@ int IDebugPortSupplierDescription2.GetDescription(enum_PORT_SUPPLIER_DESCRIPTION
             return HR.S_OK;
         }
 
-        /// <summary>Locates and loads liblinux library</summary>
-        /// <returns>True, if liblinux is successfully loaded, false otherwise</returns>
-        /// <remarks>TODO: This should go away once the credential store + connection manager
-        /// components we use are properly componentized</remarks>
-        private bool LocateAndLoadLibLinux()
+        /// <summary>
+        /// Checks if LibLinux is available by getting IVsConnectionManager service.
+        /// </summary>
+        /// <returns>True if LibLinux is available, false otherwise.</returns>
+        private bool IsLibLinuxAvailable()
         {
-            const int VSSPROPID_InstallRootDir = -9041; 
-            const string relativePath = @"Common7\IDE\CommonExtensions\Microsoft\Linux\Linux\liblinux.dll"; 
-
             IVsShell shell = Package.GetGlobalService(typeof(SVsShell)) as IVsShell;
 
             if (shell == null)
                 return false;
 
-            object pvar;
-            string libLinuxPath = null;
-            if (shell.GetProperty(VSSPROPID_InstallRootDir, out pvar) == HR.S_OK && pvar != null)
-            {
-                libLinuxPath = pvar.ToString();
-            }
-
-            libLinuxPath += relativePath;
-
-            try
-            {
-                Assembly.LoadFrom(libLinuxPath);
-            }
-            catch (Exception)
-            {
-                return false;
-            }
-
-            return true;
+            return ((IVsConnectionManager)ServiceProvider.GlobalProvider.GetService(typeof(IVsConnectionManager))) != null;
```

Comment:
Since you're just checking for null, either the `(IVsConnectionManager)` cast is unnecessary or should be an `as IVsConnectionManager` cast.